packages/block-library/src/navigation/edit/placeholder/index.js

```diff
@@ -50,30 +66,194 @@ export default function Placeholder( {
+		setMenuName( name );
+	};
+
+	const onCreateEmptyMenu = ( name ) => {
+		onFinishMenuCreation( name, [] );
+	};
+
+	const onCreateAllPages = ( name ) => {
+		const block = [ createBlock( 'core/page-list' ) ];
+		onFinishMenuCreation( name, block );
+		setIsNewMenuModalVisible( true );
+	};
+
+	useEffect( () => {
+		// If the user selected a menu but we had to wait for menu items to
+		// finish resolving, then create the block once resolution finishes.
+		if ( isCreatingFromMenu && hasResolvedMenuItems ) {
+			createFromMenu( menuName );
+			setIsCreatingFromMenu( false );
+		}
+	}, [ isCreatingFromMenu, hasResolvedMenuItems, menuName ] );
+
+	const toggleProps = {
+		variant: 'primary',
+		className: 'wp-block-navigation-placeholder__actions__dropdown',
+	};
+
+	const { navigationMenus } = useNavigationMenu();
+
 	return (
 		<>
-			{ step === PLACEHOLDER_STEPS.selectNavigationPost && (
-				<SelectNavigationMenuStep
-					onCreateNew={ ( newTitle ) => {
-						setNavigationMenuTitle( newTitle );
-						setStep( PLACEHOLDER_STEPS.createInnerBlocks );
-					} }
-					onSelectExisting={ ( navigationMenu ) => {
-						onFinish( navigationMenu );
-					} }
-					canSwitchNavigationMenu={ canSwitchNavigationMenu }
-					hasResolvedNavigationMenu={ hasResolvedNavigationMenu }
-				/>
+			{ ! hasResolvedNavigationMenu && <PlaceholderPreview isLoading /> }
+			{ hasResolvedNavigationMenu && (
+				<Placeholder className="wp-block-navigation-placeholder">
+					<PlaceholderPreview />
+
+					<div className="wp-block-navigation-placeholder__controls">
+						{ isStillLoading && (
+							<div>
+								<Spinner />
+							</div>
+						) }
+						{ ! isStillLoading && (
+							<div className="wp-block-navigation-placeholder__actions">
+								<div className="wp-block-navigation-placeholder__actions__indicator">
+									<Icon icon={ navigation } />{ ' ' }
+									{ __( 'Navigation' ) }
+								</div>
+								{ hasMenus || canSwitchNavigationMenu ? (
+									<DropdownMenu
+										text={ __( 'Add existing menu' ) }
+										icon={ chevronDown }
+										toggleProps={ toggleProps }
+										popoverProps={ { isAlternate: true } }
+									>
+										{ ( { onClose } ) => (
+											<>
+												<MenuGroup label="Menus">
+													{ canSwitchNavigationMenu &&
+														navigationMenus.map(
+															( menu ) => {
+																return (
+																	<MenuItem
+																		onClick={ () => {
+																			setSelectedMenu(
+																				menu.id
+																			);
+																			onFinish(
+																				menu
+																			);
+																		} }
+																		onClose={
+																			onClose
+																		}
+																		key={
+																			menu.id
+																		}
+																	>
+																		{
+																			menu
+																				.title
+																				.rendered
+																		}
+																	</MenuItem>
```

Comment:
This code gets very deeply nested. It's very hard to read a side-by-side diff. I think it'd be good to consider breaking the placeholder down into separate components.